### PR TITLE
Reference test suite in installation documentation

### LINKED ISSUE #8452: Documentation doesn't explain how to run tests
## Reporting a bug

- [x] I have tried using the latest released version of Numba (most recent is
 visible in the change log (https://github.com/numba/numba/blob/main/CHANGE_LOG).
- [x] I have included a self contained code sample to reproduce the problem.
  i.e. it's possible to run as 'python bug.py'.

The [installing section](https://numba.readthedocs.io/en/stable/user/installing.html) in the docs doesn't explain how to run tests.

I guessed that the script might be ```runtests.py```. But it doesn't print progress info, unlike pytest does.

Documentation should have the command that runs tests.

### PR DESCRIPTION
**Pull request by apmasell**
As titled. Closes #8452.

### CODE REVIEW

File: docs/source/user/installing.rst [hunk 1]
```diff
@@ -152,6 +152,8 @@ Then you can build and install Numba from the top level of the source tree::
 
     $ python setup.py install
 
+If you wish to run the test suite, see the :doc:`developer documentation <../developer/contributing>`.
```
Comment (stuartarchibald) [inline docs/source/user/installing.rst:155]: Please could this be line wrapped to 80 chars?

Also, if there's no anchor creating a link to the "how to run the test suite" part of the contributing docs, then perhaps it'd be a good idea to add one in there so there's a direct link that is available that can be easily shared (it also would mean this reference could jump straight to the relevant part of the docs).

Comment (stuartarchibald) [inline docs/source/user/installing.rst:155]:
```suggestion
If you wish to run the test suite, see the instructions in :doc:`developer documentation <../developer/contributing>`.
```

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch, couple of minor things to resolve else looks good.

File: docs/source/user/installing.rst (hunk 1 above)
Comment (stuartarchibald) [reply docs/source/user/installing.rst]: Thanks, I see in 79c2b3a there's no specific ref/link pair. An example of this for e.g. references and linking to the "continuous integration testing" section is:
**anchor**:
https://github.com/numba/numba/blob/061f21a73a574ea1b3451910a36db4f9b3ae788b/docs/source/developer/contributing.rst?plain=1#L325

**reference**:
https://github.com/numba/numba/blob/061f21a73a574ea1b3451910a36db4f9b3ae788b/docs/source/developer/contributing.rst?plain=1#L75

Comment (apmasell) [reply docs/source/user/installing.rst]: Sorry, didn't get pushed.

File: docs/source/user/installing.rst [hunk 2]
```diff
@@ -152,6 +152,9 @@ Then you can build and install Numba from the top level of the source tree::
 
     $ python setup.py install
 
+If you wish to run the test suite, see the instructions in
+:doc:`developer documentation <running-tests>`.
```
Comment (stuartarchibald) [inline docs/source/user/installing.rst:156]: This doesn't seem to render correctly in the RTD build, might need to be a `:ref:` ?

**Comment by stuartarchibald** [comment]
NOTE: build failures are not related to this PR. #8486 is an attempt to fix.

File: docs/source/user/installing.rst [hunk 3]
```diff
@@ -152,6 +152,9 @@ Then you can build and install Numba from the top level of the source tree::
 
     $ python setup.py install
 
+If you wish to run the test suite, see the instructions in
+:ref:`developer documentation <running-tests>`.
```
Comment (stuartarchibald) [inline docs/source/user/installing.rst:156]:
```suggestion
If you wish to run the test suite, see the instructions in the
:ref:`developer documentation <running-tests>`.
```

**Comment by stuartarchibald** [comment]
@apmasell I think mainline needs merging in to pick up the changes to the Azure CI job config for the windows build from #8486 else this PR will continue using the job config that is "broken". Once that's done builds should pass.

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch, the docs appear to be rendering correctly on inspection!

**Pull request merged**